### ISSUE TITLE
LWG 3171 not implemented for filesystem::directory_entry

### ISSUE THREAD

**Issue by jwakely** [llvm/llvm-project#53551]
```c++
#include <filesystem>
#include <iostream>

int main()
{
    std::filesystem::directory_entry e;
    std::cout << e;
}
```

This doesn't compile using libc++. [LWG 3171](https://cplusplus.github.io/LWG/issue3171) says it should.

**Comment by Quuxplusone** [comment]
https://godbolt.org/z/vvnvfW5xK
Works for me.
This was claimed fixed in b6d87773feef14d7cceeed380dd6f6c9d43fd9f0, and looking back at that commit, I don't see anything wrong-looking with it. Maybe you're just on an old version?

**Comment by jwakely** [comment]
Ah yes, I was using an odler set of headers.
